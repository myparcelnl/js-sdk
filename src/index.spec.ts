import {describe, expect, it} from 'vitest';
import * as INDEX from '@/index';

describe('module exports', () => {
  it('exposes the correct data from index.ts', async () => {
    expect.assertions(1);

    expect(INDEX).toMatchSnapshot();
  });
});
