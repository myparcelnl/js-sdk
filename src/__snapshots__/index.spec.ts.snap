// Vitest Snapshot v1

exports[`module exports > exposes the correct data from index.ts 1`] = `
{
  "AbstractClient": [Function],
  "AbstractEndpoint": [Function],
  "AbstractPrivateEndpoint": [Function],
  "AbstractPublicEndpoint": [Function],
  "ApiException": [Function],
  "BASE_URL": "https://api.myparcel.nl",
  "CARRIERS": {
    "ALL": [
      {
        "HUMAN": "Cheap Cargo",
        "ID": 3,
        "NAME": "cheapcargo",
      },
      {
        "HUMAN": "DPD",
        "ID": 4,
        "NAME": "dpd",
      },
      {
        "HUMAN": "Instabox",
        "ID": 5,
        "NAME": "instabox",
      },
      {
        "HUMAN": "DHL",
        "ID": 6,
        "NAME": "dhl",
      },
      {
        "HUMAN": "Bol.com",
        "ID": 7,
        "NAME": "bol.com",
      },
      {
        "HUMAN": "UPS",
        "ID": 8,
        "NAME": "ups",
      },
    ],
    "BOL_COM": {
      "HUMAN": "Bol.com",
      "ID": 7,
      "NAME": "bol.com",
    },
    "BOL_COM_HUMAN": "Bol.com",
    "BOL_COM_ID": 7,
    "BOL_COM_NAME": "bol.com",
    "BPOST": {
      "HUMAN": "bpost",
      "ID": 2,
      "NAME": "bpost",
    },
    "BPOST_HUMAN": "bpost",
    "BPOST_ID": 2,
    "BPOST_NAME": "bpost",
    "CHEAP_CARGO": {
      "HUMAN": "Cheap Cargo",
      "ID": 3,
      "NAME": "cheapcargo",
    },
    "CHEAP_CARGO_HUMAN": "Cheap Cargo",
    "CHEAP_CARGO_ID": 3,
    "CHEAP_CARGO_NAME": "cheapcargo",
    "DHL": {
      "HUMAN": "DHL",
      "ID": 6,
      "NAME": "dhl",
    },
    "DHL_EUROPLUS": {
      "HUMAN": "DHL Europlus",
      "ID": 11,
      "NAME": "dhleuroplus",
    },
    "DHL_EUROPLUS_HUMAN": "DHL Europlus",
    "DHL_EUROPLUS_ID": 11,
    "DHL_EUROPLUS_NAME": "dhleuroplus",
    "DHL_FOR_YOU": {
      "HUMAN": "DHL For You",
      "ID": 9,
      "NAME": "dhlforyou",
    },
    "DHL_FOR_YOU_HUMAN": "DHL For You",
    "DHL_FOR_YOU_ID": 9,
    "DHL_FOR_YOU_NAME": "dhlforyou",
    "DHL_HUMAN": "DHL",
    "DHL_ID": 6,
    "DHL_NAME": "dhl",
    "DHL_PARCEL_CONNECT": {
      "HUMAN": "DHL Parcel Connect",
      "ID": 10,
      "NAME": "dhlparcelconnect",
    },
    "DHL_PARCEL_CONNECT_HUMAN": "DHL Parcel Connect",
    "DHL_PARCEL_CONNECT_ID": 10,
    "DHL_PARCEL_CONNECT_NAME": "dhlparcelconnect",
    "DPD": {
      "HUMAN": "DPD",
      "ID": 4,
      "NAME": "dpd",
    },
    "DPD_HUMAN": "DPD",
    "DPD_ID": 4,
    "DPD_NAME": "dpd",
    "INSTABOX": {
      "HUMAN": "Instabox",
      "ID": 5,
      "NAME": "instabox",
    },
    "INSTABOX_HUMAN": "Instabox",
    "INSTABOX_ID": 5,
    "INSTABOX_NAME": "instabox",
    "POSTNL": {
      "HUMAN": "PostNL",
      "ID": 1,
      "NAME": "postnl",
    },
    "POST_NL_HUMAN": "PostNL",
    "POST_NL_ID": 1,
    "POST_NL_NAME": "postnl",
    "UPS": {
      "HUMAN": "UPS",
      "ID": 8,
      "NAME": "ups",
    },
    "UPS_HUMAN": "UPS",
    "UPS_ID": 8,
    "UPS_NAME": "ups",
  },
  "COUNTRIES": {
    "AFGHANISTAN": "AF",
    "ALAND_ISLANDS": "AX",
    "ALBANIA": "AL",
    "ALGERIA": "DZ",
    "AMERICAN_SAMOA": "AS",
    "AMERICAN_VIRGIN_ISLANDS": "VI",
    "ANDORRA": "AD",
    "ANGOLA": "AO",
    "ANGUILLA": "AI",
    "ANTARCTICA": "AQ",
    "ANTIGUA_AND_BARBUDA": "AG",
    "ARGENTINA": "AR",
    "ARMENIA": "AM",
    "ARUBA": "AW",
    "AUSTRALIA": "AU",
    "AUSTRIA": "AT",
    "AZERBAIJAN": "AZ",
    "BAHAMAS": "BS",
    "BAHRAIN": "BH",
    "BANGLADESH": "BD",
    "BARBADOS": "BB",
    "BELARUS": "BY",
    "BELGIUM": "BE",
    "BELIZE": "BZ",
    "BENIN": "BJ",
    "BERMUDA": "BM",
    "BHUTAN": "BT",
    "BOLIVIA": "BO",
    "BOSNIA_AND_HERZEGOVINA": "BA",
    "BOTSWANA": "BW",
    "BOUVET_ISLAND": "BV",
    "BRAZIL": "BR",
    "BRITISH_INDIAN_OCEAN_TERRITORY": "IO",
    "BRITISH_VIRGIN_ISLANDS": "VG",
    "BRUNEI_DARUSSALAM": "BN",
    "BULGARIA": "BG",
    "BURKINA_FASO": "BF",
    "BURUNDI": "BI",
    "CABO_VERDE": "CV",
    "CAMBODIA": "KH",
    "CAMEROON": "CM",
    "CANADA": "CA",
    "CAYMAN_ISLANDS": "KY",
    "CENTRAL_AFRICAN_REPUBLIC": "CF",
    "CHAD": "TD",
    "CHILE": "CL",
    "CHINA": "CN",
    "CHRISTMAS_ISLAND": "CX",
    "COCOS_ISLANDS": "CC",
    "COLOMBIA": "CO",
    "COMOROS": "KM",
    "CONGO_BRAZZAVILLE": "CG",
    "CONGO_KINSHASA": "CD",
    "COOK_ISLANDS": "CK",
    "COSTA_RICA": "CR",
    "COTE_D_IVOIRE": "CI",
    "CROATIA": "HR",
    "CUBA": "CU",
    "CURACAO": "CW",
    "CYPRUS": "CY",
    "CZECH_REPUBLIC": "CZ",
    "DENMARK": "DK",
    "DJIBOUTI": "DJ",
    "DOMINICA": "DM",
    "DOMINICAN_REPUBLIC": "DO",
    "DUTCH_ANTILLES": "BQ",
    "ECUADOR": "EC",
    "EGYPT": "EG",
    "EL_SALVADOR": "SV",
    "EQUATORIAL_GUINEA": "GQ",
    "ERITREA": "ER",
    "ESTONIA": "EE",
    "ESWATINI": "SZ",
    "ETHIOPIA": "ET",
    "EU": [
      "AT",
      "BE",
      "BG",
      "HR",
      "CY",
      "CZ",
      "DK",
      "EE",
      "FI",
      "FR",
      "DE",
      "GR",
      "HU",
      "IE",
      "IT",
      "LV",
      "LT",
      "LU",
      "MT",
      "NL",
      "PL",
      "PT",
      "RO",
      "SK",
      "SI",
      "ES",
      "SE",
    ],
    "FALKLAND_ISLANDS": "FK",
    "FAROE_ISLANDS": "FO",
    "FIJI": "FJ",
    "FINLAND": "FI",
    "FRANCE": "FR",
    "FRENCH_GUIANA": "GF",
    "FRENCH_POLYNESIA": "PF",
    "FRENCH_SOUTHERN_TERRITORIES": "TF",
    "GABON": "GA",
    "GAMBIA": "GM",
    "GEORGIA": "GE",
    "GERMANY": "DE",
    "GHANA": "GH",
    "GIBRALTAR": "GI",
    "GREECE": "GR",
    "GREENLAND": "GL",
    "GRENADA": "GD",
    "GUADELOUPE": "GP",
    "GUAM": "GU",
    "GUATEMALA": "GT",
    "GUERNSEY": "GG",
    "GUINEA": "GN",
    "GUINEA_BISSAU": "GW",
    "GUYANA": "GY",
    "HAITI": "HT",
    "HEARD_ISLAND_AND_MCDONALD_ISLANDS": "HM",
    "HOLY_SEE": "VA",
    "HONDURAS": "HN",
    "HONG_KONG": "HK",
    "HUNGARY": "HU",
    "ICELAND": "IS",
    "INDIA": "IN",
    "INDONESIA": "ID",
    "IRAN": "IR",
    "IRAQ": "IQ",
    "IRELAND": "IE",
    "ISLE_OF_MAN": "IM",
    "ISRAEL": "IL",
    "ITALY": "IT",
    "JAMAICA": "JM",
    "JAPAN": "JP",
    "JERSEY": "JE",
    "JORDAN": "JO",
    "KAZAKHSTAN": "KZ",
    "KENYA": "KE",
    "KIRIBATI": "KI",
    "KOREA": "KR",
    "KOSOVO": "XK",
    "KUWAIT": "KW",
    "KYRGYZSTAN": "KG",
    "LAO_PEOPLE_DEMOCRATIC_REPUBLIC": "LA",
    "LATVIA": "LV",
    "LEBANON": "LB",
    "LESOTHO": "LS",
    "LIBERIA": "LR",
    "LIBYA": "LY",
    "LIECHTENSTEIN": "LI",
    "LITHUANIA": "LT",
    "LUXEMBOURG": "LU",
    "MACAO": "MO",
    "MADAGASCAR": "MG",
    "MALAWI": "MW",
    "MALAYSIA": "MY",
    "MALDIVES": "MV",
    "MALI": "ML",
    "MALTA": "MT",
    "MARSHALL_ISLANDS": "MH",
    "MARTINIQUE": "MQ",
    "MAURITANIA": "MR",
    "MAURITIUS": "MU",
    "MAYOTTE": "YT",
    "MEXICO": "MX",
    "MICRONESIA": "FM",
    "MOLDOVA": "MD",
    "MONACO": "MC",
    "MONGOLIA": "MN",
    "MONTENEGRO": "ME",
    "MONTSERRAT": "MS",
    "MOROCCO": "MA",
    "MOZAMBIQUE": "MZ",
    "MYANMAR": "MM",
    "NAMIBIA": "NA",
    "NAURU": "NR",
    "NEPAL": "NP",
    "NETHERLANDS": "NL",
    "NEW_CALEDONIA": "NC",
    "NEW_ZEALAND": "NZ",
    "NICARAGUA": "NI",
    "NIGER": "NE",
    "NIGERIA": "NG",
    "NIUE": "NU",
    "NORFOLK_ISLAND": "NF",
    "NORTHERN_MARIANA_ISLANDS": "MP",
    "NORTH_MACEDONIA": "MK",
    "NORWAY": "NO",
    "OMAN": "OM",
    "PAKISTAN": "PK",
    "PALAU": "PW",
    "PALESTINE": "PS",
    "PANAMA": "PA",
    "PAPUA_NEW_GUINEA": "PG",
    "PARAGUAY": "PY",
    "PERU": "PE",
    "PHILIPPINES": "PH",
    "PITCAIRN": "PN",
    "POLAND": "PL",
    "PORTUGAL": "PT",
    "PUERTO_RICO": "PR",
    "QATAR": "QA",
    "REUNION": "RE",
    "ROMANIA": "RO",
    "RUSSIAN_FEDERATION": "RU",
    "RWANDA": "RW",
    "SAINT_BARTHELEMY": "BL",
    "SAINT_HELENA": "SH",
    "SAINT_KITTS_AND_NEVIS": "KN",
    "SAINT_LUCIA": "LC",
    "SAINT_MARTIN": "MF",
    "SAINT_PIERRE_AND_MIQUELON": "PM",
    "SAINT_VINCENT_AND_THE_GRENADINES": "VC",
    "SAMOA": "WS",
    "SAN_MARINO": "SM",
    "SAO_TOME_AND_PRINCIPE": "ST",
    "SAUDI_ARABIA": "SA",
    "SENEGAL": "SN",
    "SERBIA": "RS",
    "SEYCHELLES": "SC",
    "SIERRA_LEONE": "SL",
    "SINGAPORE": "SG",
    "SINT_MAARTEN": "SX",
    "SLOVAKIA": "SK",
    "SLOVENIA": "SI",
    "SOLOMON_ISLANDS": "SB",
    "SOMALIA": "SO",
    "SOUTH_AFRICA": "ZA",
    "SOUTH_GEORGIA_AND_THE_SOUTH_SANDWICH_ISLANDS": "GS",
    "SOUTH_SUDAN": "SS",
    "SPAIN": "ES",
    "SRI_LANKA": "LK",
    "SUDAN": "SD",
    "SURINAME": "SR",
    "SVALBARD_AND_JAN_MAYEN": "SJ",
    "SWEDEN": "SE",
    "SWITZERLAND": "CH",
    "SYRIAN_ARAB_REPUBLIC": "SY",
    "TAIWAN": "TW",
    "TAJIKISTAN": "TJ",
    "TANZANIA": "TZ",
    "THAILAND": "TH",
    "TIMOR_LESTE": "TL",
    "TOGO": "TG",
    "TOKELAU": "TK",
    "TONGA": "TO",
    "TRINIDAD_AND_TOBAGO": "TT",
    "TUNISIA": "TN",
    "TURKEY": "TR",
    "TURKMENISTAN": "TM",
    "TURKS_AND_CAICOS_ISLANDS": "TC",
    "TUVALU": "TV",
    "UGANDA": "UG",
    "UKRAINE": "UA",
    "UNITED_ARAB_EMIRATES": "AE",
    "UNITED_KINGDOM": "GB",
    "UNITED_STATES_MINOR_OUTLYING_ISLANDS": "UM",
    "UNITED_STATES_OF_AMERICA": "US",
    "URUGUAY": "UY",
    "UZBEKISTAN": "UZ",
    "VANUATU": "VU",
    "VENEZUELA": "VE",
    "VIETNAM": "VN",
    "WALLIS_AND_FUTUNA": "WF",
    "WESTERN_SAHARA": "EH",
    "YEMEN": "YE",
    "ZAMBIA": "ZM",
    "ZIMBABWE": "ZW",
  },
  "DELIVERY_TYPES": {
    "ALL": [
      {
        "ID": 1,
        "NAME": "morning",
      },
      {
        "ID": 2,
        "NAME": "standard",
      },
      {
        "ID": 3,
        "NAME": "evening",
      },
      {
        "ID": 4,
        "NAME": "pickup",
      },
    ],
    "EVENING": {
      "ID": 3,
      "NAME": "evening",
    },
    "EVENING_ID": 3,
    "EVENING_NAME": "evening",
    "MORNING": {
      "ID": 1,
      "NAME": "morning",
    },
    "MORNING_ID": 1,
    "MORNING_NAME": "morning",
    "PICKUP": {
      "ID": 4,
      "NAME": "pickup",
    },
    "PICKUP_ID": 4,
    "PICKUP_NAME": "pickup",
    "STANDARD": {
      "ID": 2,
      "NAME": "standard",
    },
    "STANDARD_ID": 2,
    "STANDARD_NAME": "standard",
  },
  "FetchClient": [Function],
  "GetCarrier": [Function],
  "GetCarriers": [Function],
  "GetDeliveryOptions": [Function],
  "GetPickupLocations": [Function],
  "GetShipment": [Function],
  "GetShipments": [Function],
  "PACKAGE_TYPES": {
    "ALL": [
      {
        "ID": 1,
        "NAME": "package",
      },
      {
        "ID": 2,
        "NAME": "mailbox",
      },
      {
        "ID": 3,
        "NAME": "letter",
      },
      {
        "ID": 4,
        "NAME": "digital_stamp",
      },
    ],
    "DIGITAL_STAMP": {
      "ID": 4,
      "NAME": "digital_stamp",
    },
    "DIGITAL_STAMP_ID": 4,
    "DIGITAL_STAMP_NAME": "digital_stamp",
    "LETTER": {
      "ID": 3,
      "NAME": "letter",
    },
    "LETTER_ID": 3,
    "LETTER_NAME": "letter",
    "MAILBOX": {
      "ID": 2,
      "NAME": "mailbox",
    },
    "MAILBOX_ID": 2,
    "MAILBOX_NAME": "mailbox",
    "PACKAGE": {
      "ID": 1,
      "NAME": "package",
    },
    "PACKAGE_ID": 1,
    "PACKAGE_NAME": "package",
  },
  "PLATFORMS": {
    "ALL": [
      {
        "HUMAN": "MyParcel",
        "ID": 1,
        "NAME": "myparcel",
      },
      {
        "HUMAN": "SendMyParcel",
        "ID": 2,
        "NAME": "belgie",
      },
      {
        "HUMAN": "Flespakket",
        "ID": 3,
        "NAME": "flespakket",
      },
    ],
    "FLESPAKKET": {
      "HUMAN": "Flespakket",
      "ID": 3,
      "NAME": "flespakket",
    },
    "FLESPAKKET_HUMAN": "Flespakket",
    "FLESPAKKET_ID": 3,
    "FLESPAKKET_NAME": "flespakket",
    "MYPARCEL": {
      "HUMAN": "MyParcel",
      "ID": 1,
      "NAME": "myparcel",
    },
    "MYPARCEL_HUMAN": "MyParcel",
    "MYPARCEL_ID": 1,
    "MYPARCEL_NAME": "myparcel",
    "SENDMYPARCEL": {
      "HUMAN": "SendMyParcel",
      "ID": 2,
      "NAME": "belgie",
    },
    "SENDMYPARCEL_HUMAN": "SendMyParcel",
    "SENDMYPARCEL_ID": 2,
    "SENDMYPARCEL_NAME": "belgie",
  },
  "PostShipments": [Function],
  "SHIPMENT_OPTIONS": {
    "AGE_CHECK": {
      "HUMAN": "Age check",
      "NAME": "age_check",
    },
    "AGE_CHECK_HUMAN": "Age check",
    "AGE_CHECK_NAME": "age_check",
    "ALL": [
      {
        "HUMAN": "Age check",
        "NAME": "age_check",
      },
      {
        "HUMAN": "Cooled delivery",
        "NAME": "cooled_delivery",
      },
      {
        "HUMAN": "Insurance",
        "NAME": "insurance",
      },
      {
        "HUMAN": "Large format",
        "NAME": "large_format",
      },
      {
        "HUMAN": "Only recipient",
        "NAME": "only_recipient",
      },
      {
        "HUMAN": "Printerless return",
        "NAME": "printerless_return",
      },
      {
        "HUMAN": "Return",
        "NAME": "return",
      },
      {
        "HUMAN": "Same day delivery",
        "NAME": "same_day_delivery",
      },
      {
        "HUMAN": "Signature",
        "NAME": "signature",
      },
    ],
    "COOLED_DELIVERY": {
      "HUMAN": "Cooled delivery",
      "NAME": "cooled_delivery",
    },
    "COOLED_DELIVERY_HUMAN": "Cooled delivery",
    "COOLED_DELIVERY_NAME": "cooled_delivery",
    "INSURANCE": {
      "HUMAN": "Insurance",
      "NAME": "insurance",
    },
    "INSURANCE_HUMAN": "Insurance",
    "INSURANCE_NAME": "insurance",
    "LARGE_FORMAT": {
      "HUMAN": "Large format",
      "NAME": "large_format",
    },
    "LARGE_FORMAT_HUMAN": "Large format",
    "LARGE_FORMAT_NAME": "large_format",
    "ONLY_RECIPIENT": {
      "HUMAN": "Only recipient",
      "NAME": "only_recipient",
    },
    "ONLY_RECIPIENT_HUMAN": "Only recipient",
    "ONLY_RECIPIENT_NAME": "only_recipient",
    "PRINTERLESS_RETURN": {
      "HUMAN": "Printerless return",
      "NAME": "printerless_return",
    },
    "PRINTERLESS_RETURN_HUMAN": "Printerless return",
    "PRINTERLESS_RETURN_NAME": "printerless_return",
    "RETURN": {
      "HUMAN": "Return",
      "NAME": "return",
    },
    "RETURN_HUMAN": "Return",
    "RETURN_NAME": "return",
    "SAME_DAY_DELIVERY": {
      "HUMAN": "Same day delivery",
      "NAME": "same_day_delivery",
    },
    "SAME_DAY_DELIVERY_HUMAN": "Same day delivery",
    "SAME_DAY_DELIVERY_NAME": "same_day_delivery",
    "SIGNATURE": {
      "HUMAN": "Signature",
      "NAME": "signature",
    },
    "SIGNATURE_HUMAN": "Signature",
    "SIGNATURE_NAME": "signature",
  },
  "UserException": [Function],
  "createMyParcelSdk": [Function],
  "createPrivateSdk": [Function],
  "createPublicSdk": [Function],
}
`;
